src/atoms/Text/Text.native.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components/native';
+import theme from '../../tokens/theme';
+import { Text as NativeText } from 'react-native';
+import { getAllColorNames, resolveColorFromString } from '../../utils/colors';
+import automation from '../../_helpers/automation-attributes';
+
+const fontFamilies = {
+  light: `Lato-Regular`,
+  regular: `Lato-Regular`,
+  bold: `Lato-Bold`,
+};
+
+const StyledText = styled(NativeText)(
+  (props) => `
+  font-family: ${fontFamilies[props.weight]};
+  font-size: ${props.theme.fonts.size[props.size]};
+  color: ${resolveColorFromString(props.theme, props.color) || props.theme.colors.shade[800]};
+  text-decoration-line: ${props.underline ? 'underline' : 'none'};
+  align-self: ${props.align};
+`,
+);
+
+const Text = ({ weight, size, underline, align, testID, color, children }) => {
+  return (
+    <StyledText
+      weight={weight}
+      size={size}
+      underline={underline}
+      align={align}
+      color={color}
+      {...automation(testID)}
+    >
+      {children}
+    </StyledText>
+  );
+};
+
+Text.propTypes = {
+  children: PropTypes.string,
+  weight: PropTypes.oneOf(Object.keys(theme.fonts.weight)),
+  size: PropTypes.oneOf(['xxsmall', 'xsmall', 'small', 'medium', 'large']),
+  color: PropTypes.oneOf(getAllColorNames()),
+  underline: PropTypes.bool,
```

Comment:
`is*` for booleans? so this becomes `isUnderlined`.


Also is it because link uses this that's why we expose this prop?

Reply:
I'll update it to isUnderlined. 

and yes, its because Link needs it